### DISCUSSION TITLE
Deploy volume with smaller volume

### DISCUSSION THREAD

**Discussion by OkenKhuman** [seaweedfs/seaweedfs#3891] [General]
Hi I am trying to deploy seaweed volume server with smaller volume (2GB).

I have already set the limiit in master server using `./weed master -mdir="./mdir" -volumeSizeLimitMB="2000"` and run the volume server `./weed volume -max=0 -rack="pihome" -disk="tmpfs" -mserver="localhost:9333" -dir="./ramweed"`. The volume server gets added in master server without any issue but once I try to upload files this error `rpc error: code = Unknown desc = no free volumes left for {"replication":{},"ttl":{"Count":0,"Unit":0},"preallocate":2097152000}` pops up.

Hardware:
RPi4 8GB x1 

Storage medium: tmpfs

Use version: `version 30GB 3.32 758d70bdc09834066cfd835d0041f38ac77a1d42 linux arm64`

on Status check with curl `"http://localhost:9333/dir/status?pretty=y"`
```
{
  "Topology": {
    "Max": 1,
    "Free": 1,
    "DataCenters": [
      {
        "Id": "DefaultDataCenter",
        "Racks": [
          {
            "Id": "pihome",
            "DataNodes": [
              {
                "Url": "192.168.0.12:8080",
                "PublicUrl": "192.168.0.12:8080",
                "Volumes": 0,
                "EcShards": 0,
                "Max": 1,
                "VolumeIds": " "
              }
            ]
          }
        ]
      }
    ],
    "Layouts": [
      {
        "replication": "000",
        "ttl": "",
        "writables": null,
        "collection": ""
      }
    ]
  },
  "Version": "30GB 3.32 758d70bdc09834066cfd835d0041f38ac77a1d42"
}
```

Do I need to recompile with modification or am I missing any configuration ? If I need to modify please let me know where

**Comment by chrislusf** [comment]
Added https://github.com/seaweedfs/seaweedfs/wiki/FAQ#how-many-volumes-do-i-need

You will need to reduce `volumeSizeLimitMB` size.

**Reply by OkenKhuman** [reply]
Thank you for response. I did use `volumeSizeLimitMB` (as already mention in my question).

The issue seems to resolved when I use `weed server` instead of `weed master` and `weed volume` seperately.

Now everything is working properly except for webdav speed in windows mount.

**Reply by chrislusf** [reply]
`-volumeSizeLimitMB=2000` is too large for a 8GB disk space.
